src/Sentry.DiagnosticSource/Internals/DiagnosticSource/SentryEFCoreListener.cs

```diff
@@ -75,45 +79,40 @@ public SentryEFCoreListener(IHub hub, SentryOptions options)
                     return;
                 }
 
-                if (GetSpanBucket(type) is not { } asyncLocalSpan)
-                {
-                    return;
-                }
-
+                var asyncLocalSpan = GetSpanBucket(type);
                 asyncLocalSpan.Value = new WeakReference<ISpan>(startedChild);
-                span = startedChild;
             });
-            return span;
         }
 
         private ISpan? TakeSpan(SentryEFSpanType type)
         {
             ISpan? span = null;
             _hub.ConfigureScope(scope =>
             {
-                if (scope.Transaction?.IsSampled == true)
+                if (scope.Transaction?.IsSampled != true)
                 {
-                    if (GetSpanBucket(type)?.Value is { } reference &&
-                        reference.TryGetTarget(out var startedSpan))
-                    {
-                        span = startedSpan;
-                    }
-                    else
-                    {
-                        _options.LogWarning("Trying to close a span that was already garbage collected. {0}", type);
-                    }
+                    return;
+                }
+
+                if (GetSpanBucket(type).Value is { } reference &&
+                    reference.TryGetTarget(out var startedSpan))
+                {
+                    span = startedSpan;
+                    return;
                 }
+
+                _options.LogWarning("Trying to close a span that was already garbage collected. {0}", type);
             });
             return span;
         }
 
-        private AsyncLocal<WeakReference<ISpan>>? GetSpanBucket(SentryEFSpanType type)
+        private AsyncLocal<WeakReference<ISpan>> GetSpanBucket(SentryEFSpanType type)
             => type switch
             {
                 SentryEFSpanType.QueryCompiler => _spansCompilerLocal,
                 SentryEFSpanType.QueryExecution => _spansQueryLocal,
                 SentryEFSpanType.Connection => _spansConnectionLocal,
-                _ => null
+                _ => throw new NotImplementedException()
```

Comment:
This would have silently logged a warning and now it will throw.  Is that desired?

Reply:
yep thats desired. is someone adds a value to that enum we want it to fail asap. but i will add a better exception message